chef-config/lib/chef-config/config.rb

```diff
@@ -513,7 +513,32 @@ def self.path_accessible?(path)
     default :recipe_url, nil
 
     # Set to true if Chef is to set OpenSSL to run in FIPS mode
-    default(:fips) { ENV["CHEF_FIPS"] == "1" }
+    default(:fips) do
+      !ENV["CHEF_FIPS"].nil? || check_fips_via_ohai
+    end
+
+    # we want to synchronize this ohai call because ohai is not thread safe
+    # if this gets called in a mulithreaded context, each thread's ohai instance
+    # will call reset_system while other threads are loading plugins
+    # the destructive power of reset_system is scoped to the module and not to the instance
+    def self.check_fips_via_ohai
+      @@sync ||= Mutex.new
```

Comment:
ok thats good to know. That was the one line I was most concerned about.

You mean using `@var` instead of `@@var` right. Because `default` is a class scoped method, I assumed this was the right approach.My understanding (perhaps misguided) is that `@var` is instance scoped and thus not a good fit here.